src/internal/operators/reduce.ts

```diff
@@ -54,28 +54,32 @@ export function reduce<T, R>(accumulator: (acc: R, value: T, index: number) => R
  * @see {@link mergeScan}
  * @see {@link scan}
  *
- * @param {function(acc: R, value: T, index: number): R} accumulator The accumulator function
+ * @param {function(acc: A, value: V, index: number): A} accumulator The accumulator function
  * called on each source value.
- * @param {R} [seed] The initial accumulation value.
- * @return {Observable<R>} An Observable that emits a single value that is the
+ * @param {A} [seed] The initial accumulation value.
+ * @return {Observable<A>} An Observable that emits a single value that is the
  * result of accumulating the values emitted by the source Observable.
  * @method reduce
  * @owner Observable
  */
-export function reduce<T, R>(accumulator: (acc: T | R, value: T, index?: number) => T | R, seed?: T | R): OperatorFunction<T, T | R> {
+export function reduce<V, A>(accumulator: (acc: V | A, value: V, index?: number) => A, seed?: any): OperatorFunction<V, V | A> {
```

Comment:
Does `seed` really need to be `any`? It seems to be `S` in the implementation signature for `scan`.

Reply:
Given the hoops you need to jump through in the implementation if it's `S`, I just made it `any`. This is the implementation, not the type signature.